apps/email/js/cards/cmp/autocomplete.js

```diff
@@ -0,0 +1,476 @@
+      // as typing occurs, where the parent's origin would not change, and would
+      // be a bad reference point to use.
+      var inputRect = this.inputNode.getBoundingClientRect();
+
+      // Use the size of inputArea to size the spacer to allow that area to
+      // scroll. Always recompute this because it can grow as new peep bubbles
+      // are added.
+      var inputAreaHeight = this.inputArea.getBoundingClientRect().height;
+
+      // By sizing the spacer to the same size, it allows a full scroll up
+      // of the input area.
+      this.spacer.style.minHeight = inputAreaHeight + 'px';
+
+      // Adjust the parent, which contains the inputArea and autocomplete, to
+      // only show the inputAreaHeight so that scrolling works.
+      this.parentNode.style.maxHeight = inputAreaHeight + 'px';
+
+      // Scroll the area so that the input item is now at the top of the
+      // address area.
+      var scrollTop = inputRect.top - this.originTop +
+                      this.parentNode.scrollTop -
+                      this.verticalSpace;
+
+      if (scrollTop !== this.parentNode.scrollTop) {
+        this.parentNode.scrollTop = scrollTop;
+
+        // Redo the input rect since the scroll changes the values.
+        inputRect = this.inputNode.getBoundingClientRect();
+      }
+
+      this.listTop = inputRect.bottom - this.originTop + this.verticalSpace;
+      this.list.style.top = this.listTop + 'px';
+      this.list.style.height = (window.innerHeight - this.listTop -
+                                this.originTop) + 'px';
+
+      // Set the list scrollTop to the top because previous listing could have
+      // had it scrolled.
+      this.list.scrollTop = 0;
+    },
+
+    show: function() {
+      // Set a CSS class on parentNode indicating that autocomplete is in play
+      // and could be scrolled, so the parentNode should be set explicitly to
+      // not allow scrolling, to avoid overlapping, confusing double scrollbars.
+      this.parentNode.classList.add('cmp-autocomplete-noscroll');
+
+      // The originNode should be at the top of its scroll area, and set a
+      // CSS class on it that indicates the origin should not show scroll
+      // bars or allow scrolling, to avoid overlapping, confusing double
+      // scrollbars.
+      this.originNode.scrollTop = 0;
+      this.originNode.classList.add('cmp-autocomplete-noscroll');
+
+      this.sizeList();
+      this.list.classList.remove('collapsed');
+    },
+
+    hide: function() {
+      this.hideList();
+
+      this.parentNode.style.maxHeight = '';
+      this.parentNode.scrollTop = 0;
+
+      this.parentNode.classList.remove('cmp-autocomplete-noscroll');
+      this.originNode.classList.remove('cmp-autocomplete-noscroll');
+
+      this.spacer.style.minHeight = '';
+
+      this._matchClickedTime = 0;
+    },
+
+    hideList: function() {
+      this.list.classList.add('collapsed');
```

Comment:
Uh, noting that I didn't see if this has other horrible side-effects.  Maybe more things should just be calling hide() instead or what not.